src/ints/int10_vesa.cpp

```diff
@@ -236,29 +249,30 @@ Bit8u VESA_GetSVGAModeInformation(Bit16u mode,Bit16u seg,Bit16u off) {
 	} else if (pageSize) {
 		pages = (vga.vmemsize / pageSize)-1;
 	}
-	var_write(&minfo.NumberOfImagePages, pages);
-	var_write(&minfo.ModeAttributes, modeAttributes);
-	var_write(&minfo.WinAAttributes, 0x7);	// Exists/readable/writable
+	minfo.NumberOfImagePages = host_to_le(static_cast<uint8_t>(pages));
+	minfo.ModeAttributes = host_to_le(static_cast<uint16_t>(modeAttributes));
+	minfo.WinAAttributes = host_to_le(static_cast<uint8_t>(0x7)); // Exists/readable/writable
 
 	if (mblock->type==M_TEXT) {
-		var_write(&minfo.WinGranularity,32);
-		var_write(&minfo.WinSize,32);
-		var_write(&minfo.WinASegment,0xb800);
-		var_write(&minfo.XResolution,mblock->twidth);
-		var_write(&minfo.YResolution,mblock->theight);
+		minfo.WinGranularity = host_to_le(static_cast<uint16_t>(32));
+		minfo.WinSize = host_to_le(static_cast<uint16_t>(32));
+		minfo.WinASegment = host_to_le(static_cast<uint16_t>(0xb800));
+		minfo.XResolution = host_to_le(static_cast<uint16_t>(mblock->twidth));
+		minfo.YResolution = host_to_le(static_cast<uint16_t>(mblock->theight));
```

Comment:
OK, this commit is the one I referred to in PR introducing `include/byteorder.h`. I think it would be beneficial to avoid `host_to_le` + `static_cast` in here and use `host_to_le16` *or* `static_cast<uint8_t>(X)` (depending on field being filled).

So this piece of code would look like this:

```suggestion
		minfo.WinGranularity = host_to_le16(32);
		minfo.WinSize = host_to_le16(32);
		minfo.WinASegment = host_to_le16(0xb800);
		minfo.XResolution = host_to_le(static_cast<uint16_t>(mblock->twidth));
		minfo.YResolution = host_to_le(static_cast<uint16_t>(mblock->theight));
```

While the fields that are `uint8_t` could be either:
```
minfo.WinAAttributes = static_cast<uint8_t>(0x7); // Exists/readable/writable
```
or maybe even:
```
minfo.WinAAttributes = uint8_t(0x7); // Exists/readable/writable
```

Reply:
Also, when I was toying with a similar change I remember this code being a bit tricky with types. Go ahead and write it the way it should be written (or maybe left as it is), afterwards I will double-check if types in `MODE_INFO` match the types being written-in.

Reply:
Worked pefectly! Thanks for seeing this opportunity and whipping up a very nice drop-in solution. It massively cleans up this code (now two rounds of scrubbing).